flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherGateway.java

```diff
@@ -73,7 +74,7 @@
      *     FlinkJobNotFoundException}
      */
     @Override
-    CompletableFuture<ArchivedExecutionGraph> requestJob(JobID jobId, @RpcTimeout Time timeout);
+    CompletableFuture<ExecutionGraphInfo> requestJob(JobID jobId, @RpcTimeout Time timeout);
```

Comment:
It might be a bit easier to not change the return type of this method and instead to introduce a new method `CompletableFuture<ExecutionGraphInfo> requestJobInfo()`. Then, one could even add a default implementation of this method to call `requestJobInfo().thenApply(ExecutionGraphInfo::getArchivedExecutionGraph)`.